src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ShoppingCartService;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shoppingCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        List<Product> products = shoppingCart.getProducts();
+        products.add(product);
+        shoppingCart.setProducts(products);
```

Comment:
Do customers create products while completing an order or do they chose from the existing ones?

Reply:
I do not create any other products. I just add them to the list and update it shoppingCart.